File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/recipes/domain/usecase/RecipeCatalogUseCaseTest.java
package com.tarbadev.witchcraft.recipes.domain.usecase;

import com.tarbadev.witchcraft.recipes.domain.usecase.RecipeCatalogUseCase;
import com.tarbadev.witchcraft.recipes.domain.entity.Recipe;
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.Arrays;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.BDDMockito.given;

@RunWith(MockitoJUnitRunner.class)
@SpringBootTest
public class RecipeCatalogUseCaseTest {

    @Mock private RecipeRepository recipeRepository;

    private RecipeCatalogUseCase subject;

    @Before
    public void setUp() {
        subject = new RecipeCatalogUseCase(recipeRepository);
    }

    @Test
    public void execute() {
        List<Recipe> recipes = Arrays.asList(
                Recipe.builder().build(),
                Recipe.builder().build(),
                Recipe.builder().build()
        );

        given(recipeRepository.findAll()).willReturn(recipes);

        assertThat(subject.execute()).isEqualTo(recipes);
    }
}

File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/recipes/domain/usecase/RecipeCatalogUseCaseTest.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import com.tarbadev.witchcraft.recipes.domain.entity.Recipe
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.util.*

class RecipeCatalogUseCaseTest {
    private val recipeRepository: RecipeRepository = mock()
    private lateinit var recipeCatalogUseCase: RecipeCatalogUseCase

    @BeforeEach
    fun setUp() {
        recipeCatalogUseCase = RecipeCatalogUseCase(recipeRepository)
    }

    @Test
    fun execute() {
        val recipes = Arrays.asList(
            Recipe(),
            Recipe(),
            Recipe()
        )

        whenever(recipeRepository.findAll()).thenReturn(recipes)

        assertEquals(recipes, recipeCatalogUseCase.execute())
    }
}